c7n/resources/s3.py

```diff
@@ -2595,11 +2595,17 @@ def process_bucket(self, bucket):
                 if rule['Status'] != 'absent':
                     config.append(rule)
 
-        # The extra `list` conversion if required for python3
+        # The extra `list` conversion is required for python3
         config = list(filter(None, config))
 
-        s3.put_bucket_lifecycle_configuration(
-            Bucket=bucket['Name'], LifecycleConfiguration={'Rules': config})
+        try:
+            s3.put_bucket_lifecycle_configuration(
+                Bucket=bucket['Name'], LifecycleConfiguration={'Rules': config})
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'AccessDenied':
+                log.warning("Access Denied Bucket:%s while applying lifecycle" % bucket['Name'])
+            else:
+                raise e
```

Comment:
Is this appropriate to log and skip over any buckets that cannot be accessed?

Reply:
yes thats appropriate, ideally is if we can return a set of those buckets names on denied {'denied': [bucket_names]}, action results are serialized as a json output file.

Reply:
The error catching is handled in process. The reason it isn't working is because `.exception` is called before `.result`.   `process` needs to be changed to properly handle exceptions so that it doesn't error out when a single bucket has an exception.

Reply:
Ok I see, exception and result can't be used together.  Exception will return return if an exception is raised, however, that does not stop result from also raising the exception if result is called later.

Reply:
@kapilt - is it appropriate to change the lifecycle action to be a subclass of `ScanBucket`?  That will work with the denied_buckets.  Otherwise I can move the denied_buckets logic up a level into `BucketActionBase`

Reply:
scanbucket is for actions that are looking at objects, so it wouldn't be appropriate. simply returning a dictionary from process on the action method would suffice.